test/parquet/test_parquet_schema.test

```diff
@@ -0,0 +1,169 @@
+statement error
+SELECT *
+FROM read_parquet('__TEST_DIR__/integers.parquet', schema=map {
+                    0: {name: 'cool_column', type: 'BIGINT', default_value: NULL},
+                    1: {name: 'cool_column', type: 'UTINYINT', default_value: 'bla'}
+                  }) pq
+----
+Binder Error: Unable to cast Parquet schema default_value "bla" to UTINYINT
+
+query IIIIII
+DESCRIBE SELECT *
+FROM read_parquet('__TEST_DIR__/integers.parquet', schema=map {
+                    0: {name: 'renamed_i', type: 'BIGINT', default_value: NULL},
+                    1: {name: 'new_column', type: 'UTINYINT', default_value: 43}
+                  })
+----
+renamed_i	BIGINT	YES	NULL	NULL	NULL
+new_column	UTINYINT	YES	NULL	NULL	NULL
+
+query IIIIII
+DESCRIBE SELECT *
+FROM read_parquet('__TEST_DIR__/integers.parquet', schema=map {
+                    0: {name: 'renamed_i', type: 'BIGINT', default_value: NULL},
+                    1: {name: 'new_column', type: 'UTINYINT', default_value: 43}
+                  }, filename=true)
+----
+renamed_i	BIGINT	YES	NULL	NULL	NULL
+new_column	UTINYINT	YES	NULL	NULL	NULL
+filename	VARCHAR	YES	NULL	NULL	NULL
+
+# we'll test if filename works on a persistent file otherwise __TEST_DIR__ will be different every time
+query II
+SELECT *
+FROM read_parquet('data/parquet-testing/enum.parquet', schema=map {
+                    1: {name: 'cool_column', type: 'VARCHAR', default_value: NULL}
+                  }, filename=true)
+LIMIT 1
+----
+1	data/parquet-testing/enum.parquet
+
+query II
+SELECT *
+FROM read_parquet('__TEST_DIR__/integers.parquet', schema=map {
+                    0: {name: 'renamed_i', type: 'BIGINT', default_value: NULL},
+                    1: {name: 'new_column', type: 'UTINYINT', default_value: 43}
+                  })
+----
+42	43
+
+# we just get a cast error when we can't cast to the supplied type
+statement error
+SELECT *
+FROM read_parquet('__TEST_DIR__/integers.parquet', schema=map {
+                    0: {name: 'renamed_i', type: 'DATE', default_value: NULL}
+                  })
+----
+Conversion Error
+
+# if we don't supply a field id, we can't refer to it using the schema parameter
+statement ok
+COPY (SELECT 42::INTEGER i) TO '__TEST_DIR__/integers.parquet'
+
+query II
+SELECT *
+FROM read_parquet('__TEST_DIR__/integers.parquet', schema=map {
+                    0: {name: 'renamed_i', type: 'BIGINT', default_value: NULL},
+                    1: {name: 'new_column', type: 'UTINYINT', default_value: 43}
+                  })
+----
+NULL	43
+
+# let's spice it up with more columns
+statement ok
+COPY (
+    SELECT 1 i1, 3 i3, 4 i4, 5 i5
+) TO '__TEST_DIR__/integers.parquet' (FIELD_IDS {i1: 5, i3: 3, i4: 2, i5: 1})
+
+# this is purposely a bit confusing but we're:
+# 1. deleting field id 2
+# 2. creating field id 4
+# 3. reversing the order of the columns
+# 4. renaming them (except i3)
+# 5. upcasting them
+query IIII
```

Comment:
Maybe we can expand/add a test here that tests that it also works with multiple parquet files (but without hive partitioning)?
something like: 
```
statement ok
COPY (
    SELECT 1 i1, 3 i3, 4 i4, 5 i5
) TO '__TEST_DIR__/multifile1.parquet' (FIELD_IDS {i1: 5, i3: 3, i4: 2, i5: 1})

statement ok
COPY (
    SELECT 1 i1, 3 i3, 4 i4, 5 i5
) TO '__TEST_DIR__/multifile2.parquet' (FIELD_IDS {i1: 1, i3: 2, i4: 3, i5: 4})

query IIII
SELECT *
FROM read_parquet('__TEST_DIR__/multifile*.parquet', schema=map {
                    1: {name: 'i1', type: 'BIGINT', default_value: NULL},
                    3: {name: 'i3', type: 'BIGINT', default_value: NULL},
                    4: {name: 'i4', type: 'BIGINT', default_value: 2},
                    5: {name: 'i5', type: 'BIGINT', default_value: NULL}
                  })
ORDER BY i1
----
1	4	5	NULL
5	3	2	1
```